backend/src/v5/utils/quota.js

```diff
@@ -15,38 +15,50 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 const { createResponseCode, templates } = require('./responseCodes');
+const { getAllUsersInTeamspace, getSubscriptions } = require('../models/teamspaces');
 const DBHandler = require('../handler/db');
 const config = require('./config');
-const { getSubscriptions } = require('../models/teamspaces');
+const { getInvitationsByTeamspace } = require('../models/invitations');
 const { getTotalSize } = require('../models/fileRefs');
 
 const Quota = {};
 
-const calculateQuota = async (teamspace) => {
+Quota.getQuotaInfo = async (teamspace, inMegabytes = false) => {
 	const subs = await getSubscriptions(teamspace);
 	let quotaSize = 0;
+	let collaboratorLimit = config.subscriptions?.basic?.collaborators ?? 0;
 	let hasExpiredQuota = false;
 
-	Object.keys(subs).forEach((key) => {
-		// paypal subs have a different schema - and no oen should have an active paypal sub. Skip.
-		if (key !== 'paypal') {
-			const { expiryDate, data } = subs[key];
+	for (let i = 0; i < subs.length; i++) {
+		const sub = subs[i];
 
-			if (expiryDate && expiryDate < Date.now()) {
-				hasExpiredQuota = true;
-			} else {
-				quotaSize += data;
+		// eslint-disable-next-line no-loop-func
+		Object.keys(sub).forEach((key) => {
+			// paypal subs have a different schema - and no oen should have an active paypal sub. Skip.
+			if (key !== 'paypal') {
+				const { expiryDate, data } = sub[key];
+				if (expiryDate && expiryDate < Date.now()) {
+					hasExpiredQuota = true;
+				} else {
+					quotaSize += data;
+					if (collaboratorLimit !== 'unlimited') {
+						const subCollaborators = sub[key].collaborators;
+						collaboratorLimit = subCollaborators === 'unlimited' ? 'unlimited' : collaboratorLimit + subCollaborators;
+					}
+				}
 			}
-		}
-	});
+		});
+	}
 
 	if (hasExpiredQuota && quotaSize === 0) throw templates.licenceExpired;
 
 	const basicQuota = config.subscriptions?.basic?.data;
-	return (quotaSize + basicQuota) * 1024 * 1024; // data in MB, returning in Bytes.
+	const quotaInBytes = (quotaSize + basicQuota);
+
+	return { quota: inMegabytes ? quotaInBytes : quotaInBytes * 1024 * 1024, collaboratorLimit };
```

Comment:
both collaborator and data are part of the quota, so it's probably better to name it as that
```suggestion
	return { data: inMegabytes ? quotaInBytes : quotaInBytes * 1024 * 1024, collaborators: collaboratorLimit };
```